src/filesystem/impls/filer/FilerFileSystem.js

```diff
@@ -180,34 +180,25 @@ define(function (require, exports, module) {
         oldPath = decodePath(oldPath);
         newPath = decodePath(newPath);
 
-        function updateBlobURL(err) {
+        function updateURL(err) {
             if(err) {
                 return callback(_mapError(err));
             }
 
-            // If this was a rename on a file path, update the Blob cache too
-            stat(newPath, function(err, stat) {
+            UrlCache.rename(oldPath, newPath, function(err, removed) {
                 if(err) {
                     return callback(_mapError(err));
                 }
 
-                if(stat.isFile) {
-                    UrlCache.rename(oldPath, newPath, function(err) {
-                        if(err) {
-                            return callback(_mapError(err));
-                        }
+                removed.forEach(function(pathInfo) {
+                    BrambleEvents.triggerFileRenamed(pathInfo.oldPath, pathInfo.newPath);
```

Comment:
I *think* that what we did before was incorrect, and we should have been triggering for every path that got renamed.  Having said that, we seem to have a special case for folders here: https://github.com/mozilla/thimble.mozilla.org/blob/e3217b471f28a7dd6af53b08a028160a95fb6769/public/editor/scripts/editor/js/fc/filesystem-sync.js#L80-L81.  Can you help me figure this out?  It seems wrong to not trigger renames, but I agree, let's not break Thimble.